app/jobs/regular/mass_award_badge.rb

```diff
@@ -8,15 +8,39 @@ def execute(args)
 
       users = User.select(:id, :username, :locale)
 
-      if mode == 'email'
+      email_mode = mode == 'email'
+      if email_mode
+        args[:users_batch].map!(&:downcase)
         users = users.with_email(args[:users_batch])
       else
-        users = users.where(username_lower: args[:users_batch].map!(&:downcase))
+        args[:users_batch].map! { |u| User.normalize_username(u) }
+        users = users.where(username_lower: args[:users_batch])
       end
 
-      return if users.empty? || badge.nil?
+      return if users.empty? || badge.nil? || !badge.enabled?
```

Comment:
If the badge is `nil` or not enabled, I think we can return much earlier in the job instead of after fetching the users.